presto-main/src/main/java/com/facebook/presto/execution/QueryManagerConfig.java

```diff
@@ -159,6 +161,20 @@ public QueryManagerConfig setMaxQueryHistory(int maxQueryHistory)
         return this;
     }
 
+    @Min(1000)
```

Comment:
@martint

I want to avoid the value is set too small that doesn't make any sense (e.g. -1, 0, 10). I feel a length limit of 1000 could allow some reasonable queries.

The average length of Hive query in Warehouse is around 3000: https://fburl.com/i09f5uql .